src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -58,6 +60,62 @@ public static FormattableString GetIdentifiersFormattable(this IEnumerable<strin
                 _ => FormattableStringFactory.Create(GetNamesForMethodCallFormat(count, 'I'), identifiers.ToArray<object>())
             };
 
+        public static FormattableString GetConstantFormattable(this Constant constant)
+        {
+            if (constant.Value == null)
+            {
+                // Cast helps the overload resolution
+                return $"({constant.Type}){null:L}";
+            }
+
+            if (constant.IsNewInstanceSentinel)
+            {
+                return $"new {constant.Type}()";
+            }
+
+            if (!constant.Type.IsFrameworkType && constant.Value is EnumTypeValue enumTypeValue)
+            {
+                return $"{constant.Type}.{enumTypeValue.Declaration.Name}";
+            }
+
+            if (!constant.Type.IsFrameworkType && constant.Value is string enumValue)
+            {
+                return $"new {constant.Type}({enumValue:L})";
+            }
+
+            FormattableString result;
+            Type frameworkType = constant.Type.FrameworkType;
+            if (frameworkType == typeof(DateTimeOffset))
+            {
+                var d = (DateTimeOffset)constant.Value;
+                d = d.ToUniversalTime();
+                result = $"new {typeof(DateTimeOffset)}({d.Year:L}, {d.Month:L}, {d.Day:L} ,{d.Hour:L}, {d.Minute:L}, {d.Second:L}, {d.Millisecond:L}, {typeof(TimeSpan)}.{nameof(TimeSpan.Zero)})";
+            }
+            else if (frameworkType == typeof(byte[]))
+            {
+                result = $"new byte[] {{";
+                var value = (byte[])constant.Value;
+                foreach (byte b in value)
+                {
+                    result = $"{result}{b}, ";
+                }
+
+                result = $"{result}}}";
+                return result;
```

Comment:
Since we know the type of argument upfront (it is `byte`), we can create string upfront:

```cs
var bytes = (byte[])constant.Value;
var joinedBytes = string.Join(", ", bytes);
return $"new byte[] {{{joinedBytes}}}";
```